src/command_modules/azure-cli-servicebus/azure/cli/command_modules/servicebus/_params.py

```diff
@@ -0,0 +1,170 @@
+    key_arg_type = CLIArgumentType(options_list=['--key-name'], arg_type=get_enum_type(KeyType), help='specifies Primary or Secondary key needs to be reset')
+
+    with self.argument_context('servicebus') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('namespace_name', options_list=['--namespace-name'], help='name of the Namespace')
+
+    with self.argument_context('servicebus namespace create') as c:
+        c.argument('namespace_name', options_list=['--name', '-n'], help='name of the Namespace')
+        c.argument('tags', arg_type=tags_type)
+        c.argument('sku', options_list=['--sku-name'], arg_type=get_enum_type(SkuName))
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('skutier', options_list=['--sku-tier'], arg_type=get_enum_type(SkuTier))
+        c.argument('capacity', options_list=['--capacity'], help='Capacity for Sku')
+
+    # region Namespace Get
+    for scope in ['servicebus namespace show', 'servicebus namespace delete']:
+        with self.argument_context(scope) as c:
+            c.argument('namespace_name', options_list=['--name', '-n'], help='name of the Namespace')
+
+    # region Namespace Authorizationrule
+    with self.argument_context('servicebus namespace authorizationrule') as c:
+        c.argument('authorization_rule_name', options_list=['--name', '-n'], help='name of the Namespace AuthorizationRule')
+
+    with self.argument_context('servicebus namespace authorizationrule create') as c:
+        c.argument('accessrights', arg_type=rights_arg_type)
+
+    with self.argument_context('servicebus namespace authorizationrule keys renew') as c:
+        c.argument('key_type', arg_type=key_arg_type)
+
+    # region Queue
+    with self.argument_context('servicebus queue') as c:
+        c.argument('queue_name', options_list=['--name', '-n'], help='Name of Queue')
+
+    # region - Queue Create
+    with self.argument_context('servicebus queue create') as c:
+        c.argument('lock_duration', options_list=['--lock-duration'], validator=_validate_lock_duration, help='String ISO 8601 timespan duration of a peek-lock; that is, the amount of time that the message is locked for other receivers. The maximum value for LockDuration is 5 minutes; the default value is 1 minute.')
+        c.argument('max_size_in_megabytes', options_list=['--max-size-in-megabytes'], type=int, choices=[1024, 2048, 3072, 4096, 5120], help='The maximum size of the queue in megabytes, which is the size of memory allocated for the queue. Default is 1024.')
+        c.argument('requires_duplicate_detection', options_list=['--requires-duplicate-detection'], action='store_true', help='A boolean value indicating if this queue requires duplicate detection.')
+        c.argument('requires_session', options_list=['--requires-session'], action='store_true', help='A boolean value that indicates whether the queue supports the concept of sessions.')
```

Comment:
If `requires_session` is a Boolean property on a resource then this should not be `action='store_true'` but `arg_type=get_three_state_flag()`.

Reply:
implemented